cmd/agentctl/cli/cli.go

```diff
@@ -187,14 +189,21 @@ func (cli *AgentCli) Initialize(opts *ClientOptions, ops ...InitializeOpt) error
 	return nil
 }
 
-func newAPIClient(opts *ClientOptions) (client.APIClient, error) {
+func buildClientOptions() []client.Opt {
 	clientOpts := []client.Opt{
 		client.WithHost(viper.GetString("host")),
 		client.WithServiceLabel(viper.GetString("service-label")),
 		client.WithGrpcPort(viper.GetInt("grpc-port")),
 		client.WithHTTPPort(viper.GetInt("http-port")),
-		client.WithEtcdEndpoints(opts.Endpoints),
 	}
+
+	// Handle properly case when `etcd-endpoints` returned from environment variable
+	etcdEndp := viper.GetStringSlice("etcd-endpoints")
+	if len(etcdEndp) == 1 && strings.Contains(etcdEndp[0], ",") {
+		etcdEndp = strings.Split(etcdEndp[0], ",")
+	}
+	clientOpts = append(clientOpts, client.WithEtcdEndpoints(etcdEndp))
```

Comment:
Maybe don't allow multiple endpoints in environment variable and rename from `ETCD_ENDPOINTS` to `ETCD_ENDPOINT` when binding?

Reply:
I would keep both though for backwards compatibility since we already use `ETCD_ENDPOINTS`.